src/cargo/util/toml_mut/manifest.rs

```diff
@@ -409,6 +411,49 @@ impl LocalManifest {
                 }
             }
         }
+
+        // continue only if the dep is fully removed
+        if status != DependencyStatus::None {
+            return;
+        }
+
+        // clean up profile sections
+        let profile_table_path = [
+            "profile".to_owned(),
+            "dev".to_owned(),
```

Comment:
I assume this needs to apply to any profile, not just `dev`?

Reply:
Ah yes... I just misread my notes. Will fix.